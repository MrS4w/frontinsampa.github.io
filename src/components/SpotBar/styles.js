import { fade } from '@material-ui/core/styles/colorManipulator';

export default ({ palette }) => ({
  spotBar: {
    cursor: 'pointer',
    height: 'inherit',
    left: 0,
    position: 'absolute',
    top: 0,
    width: 'inherit',
    zIndex: 2,

    '&.actived': {
      cursor: 'default',

      '& > $spotBarContent': {
        transform: 'translateY(0)',
      },
    },
  },
  spotBarContent: {
    alignItems: 'flex-end',
    backgroundColor: fade(palette.tar, 0.8),
    display: 'flex',
    height: 'inherit',
    left: 0,
    position: 'absolute',
    top: 0,
    transform: 'translateY(100%)',
    transition: 'transform .2s',
    width: 'inherit',
  },
});
