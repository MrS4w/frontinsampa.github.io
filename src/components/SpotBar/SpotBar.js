import React from 'react';
import PropTypes from 'prop-types';

import Box from '@material-ui/core/Box';

import { withStyles } from '../../styles';

import styles from './styles';

const SpotBar = ({ classes, children }) => {
  const handleFocus = ({ target }) => {
    target.classList.add('actived');
  };

  const handleBlur = ({ target }) => {
    target.classList.remove('actived');
  };

  return (
    <div className={classes.spotBar} tabIndex="0" onFocus={handleFocus} onBlur={handleBlur} role="textbox">
      <Box className={classes.spotBarContent} padding={3}>
        <div>
          { children }
        </div>
      </Box>
    </div>
  );
};

SpotBar.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired,
};

export default withStyles(styles)(SpotBar);
