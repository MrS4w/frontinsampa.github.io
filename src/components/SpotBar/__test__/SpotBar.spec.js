import React from 'react';
import { shallow } from 'enzyme';

import SpotBar from '../SpotBar';

jest.mock('@material-ui/core/Box');

/**
 * @todo Resolve problem with `fade`.
 */
jest.mock('../styles', () => ({
  spotBar: {},
  spotBarContent: {},
}));

describe('Components: <SpotBar />', () => {
  const wrapper = shallow(
    <SpotBar classes={{}}>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </SpotBar>,
  );

  it('Create a snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
