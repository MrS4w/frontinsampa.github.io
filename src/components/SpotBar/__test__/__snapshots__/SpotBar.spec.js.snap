// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <SpotBar /> Create a snapshot. 1`] = `
<SpotBar
  classes={
    Object {
      "spotBar": "SpotBar-spotBar-1",
      "spotBarContent": "SpotBar-spotBarContent-2",
    }
  }
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</SpotBar>
`;
